### PR TITLE
Z: Change CPU support flag setters to return void

### PR DESCRIPTION
**Pull request by Spencer-Comin**
Currently the `OMR::Z::CPU::setSupports*(bool value)` methods return the input `value`. The return value from these setters is not used anywhere, so there is no need to have a return value.

### CODE REVIEW

**Review by r30shah — APPROVED** [review]
LGTM

**Comment by r30shah** [comment]
@Spencer-Comin  can you resolve the conflicts this is complaining about so that I can launch tests?

**Comment by r30shah** [comment]
Jenkins build zos,zlinux

**Review by r30shah — APPROVED** [review]

**Comment by r30shah** [comment]
Jenkins build zos,zlinux

File: compiler/z/env/OMRCPU.cpp [hunk 1]
```diff
@@ -266,10 +266,10 @@ OMR::Z::CPU::getSupportsArch(Architecture arch)
    return _supportedArch >= arch;
    }
 
-bool
+void
 OMR::Z::CPU::setSupportsArch(Architecture arch)
    {
-   return _supportedArch = _supportedArch >= arch ? _supportedArch : arch;
+   _supportedArch = _supportedArch >= arch ? _supportedArch : arch;
    }
```
Comment (hzongaro) [inline compiler/z/env/OMRCPU.cpp:273]: This `setSupportsArch` method seems a bit odd to me.  This is one where returning a boolean value to indicate whether the method had an effect might make sense, although it's currently returning something that will always be implicitly cast to true, I think.  I see it's never actually used, so perhaps it should be removed?

**Comment by r30shah** [comment]
Jenkins build zos,zlinux

**Review by hzongaro — COMMENTED** [review]
I have a question/comment about `setSupportsArch`

File: compiler/z/env/OMRCPU.cpp (hunk 1 above)
Comment (Spencer-Comin) [reply compiler/z/env/OMRCPU.cpp:273]: Yes, looks like I just forgot to remove the `return`

Comment (hzongaro) [reply compiler/z/env/OMRCPU.cpp:273]: No, you had removed the `return`.  What I find odd is that the method might not do what is requested.  For instance, if I try to call `setSupportsArch(z9)` when the default `z10` is in effect, the method won't do anything.  Having it return a boolean that indicates whether the method had an effect might be useful.

But as I said, the method is never used, so maybe it should be removed.

**Review by hzongaro — APPROVED** [review]
Looks good.  Thanks!

**Comment by hzongaro** [comment]
Jenkins build zos,zlinux

**Pull request merged**